File: bin/agente.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-
import sys; sys.path.append('../lib')
import Reglas

class MiAgente (Reglas.Agente):

    def __init__(self):
        Reglas.Agente.__init__(self)
        
    def iniciar(self, Tablero, id):
        self.tab = Tablero
        self.id = id
        print "Estoy iniciado"
        
    def siguienteJugada(self):
        print self.id, ": Preparo mi jugada"
        jugada = Reglas.Jugada()
        jugada.setPosicion(4,5)
        jugada.setTipoDeJugada(Reglas.TipoDeJugada.MOVIMIENTO)
        
        for meta in self.tab.getMetas(self.id):
            print meta
        
        return jugada
    
    def terminar(self):
        print "He terminado", self.id

File: src/AgenteWrapper/Agente.cpp
#include <pybind11/pybind11.h>
#include <Reglas/Agente.hpp>

namespace py = pybind11;
using namespace Reglas;

struct PyAgente : Agente {
    using Agente::Agente;
    Jugada siguienteJugada() override {
        PYBIND11_OVERRIDE_PURE(Jugada, Agente, siguienteJugada);
    }
    void iniciar(int id) override {
        PYBIND11_OVERRIDE_PURE(void, Agente, iniciar, id);
    }
    void terminar() override {
        PYBIND11_OVERRIDE_PURE(void, Agente, terminar);
    }
};

void export_agente(py::module_& m)
{
    py::class_<Agente, PyAgente>(m, "Agente")
        .def(py::init<>())
        .def("siguienteJugada", &Agente::siguienteJugada)
        .def("iniciar", &Agente::iniciar)
        .def("terminar", &Agente::terminar);
}
